docs/src/background/infeasibility_certificates.md

```diff
@@ -0,0 +1,132 @@
+    const MOI = MathOptInterface
+end
+DocTestFilters = [r"MathOptInterface|MOI"]
+```
+
+# Infeasibility certificates
+
+When given an infeasible or unbounded model, some (conic) solvers can produce a
+certificate or infeasibility. This page explains what a certificate of
+infeasibility is for unbounded and infeasible models.
+
+## Conic duality
+
+MathOptInterface uses conic duality, which we use to define infeasibility
+certificates. A full explanation is given in the section [Duality](@ref), but
+here is a brief overview.
+
+For a minimization problem in geometric conic form, the primal is:
+```math
+\begin{align}
+& \min_{x \in \mathbb{R}^n} & a_0^\top x + b_0
+\\
+& \;\;\text{s.t.} & A_i x + b_i & \in \mathcal{C}_i & i = 1 \ldots m,
+\end{align}
+```
+and the dual is a maximization problem in standard conic form:
+```math
+\begin{align}
+& \max_{y_1, \ldots, y_m} & -\sum_{i=1}^m b_i^\top y_i + b_0
+\\
+& \;\;\text{s.t.} & a_0 - \sum_{i=1}^m A_i^\top y_i & = 0
+\\
+& & y_i & \in \mathcal{C}_i^* & i = 1 \ldots m,
+\end{align}
+```
+where each ``\mathcal{C}_i`` is a closed convex cone and ``\mathcal{C}_i^*`` is
+its dual cone.
+
+For a maximization problem in geometric conic form, the primal is:
+```math
+\begin{align}
+& \max_{x \in \mathbb{R}^n} & a_0^\top x + b_0
+\\
+& \;\;\text{s.t.} & A_i x + b_i & \in \mathcal{C}_i & i = 1 \ldots m,
+\end{align}
+```
+and the dual is a minimization problem in standard conic form:
+```math
+\begin{align}
+& \min_{y_1, \ldots, y_m} & \sum_{i=1}^m b_i^\top y_i + b_0
+\\
+& \;\;\text{s.t.} & a_0 + \sum_{i=1}^m A_i^\top y_i & = 0
+\\
+& & y_i & \in \mathcal{C}_i^* & i = 1 \ldots m.
+\end{align}
+```
+
+## Unbounded models
+
+If a model is unbounded, two things are true:
+ 1. there exists a feasible primal solution
+ 2. the dual is infeasible.
+
+A feasible primal solution---if one exists---can be obtained by setting
+[`ObjectiveSense`](@ref) to `FEASIBILITY_SENSE` and then optimizing. Therefore,
+most solvers terminate after they prove the dual is infeasible via a certificate
+of dual infeasibility, but _before_ they have found a feasible primal solution.
+This is also the reason that MathOptInterface defines the `DUAL_INFEASIBLE`
+status instead of `UNBOUNDED`.
+
+A certificate of dual infeasibility is an improving ray of the primal problem.
+That is, there exists some vector ``d`` such that for all ``\eta > 0``:
+```math
+A_i (x + \eta d) + b_i \in \mathcal{C}_i,\ \ i = 1 \ldots m,
+```
+and (for minimization problems):
+```math
+a_0^\top (x + \eta d) + b_0 < a_0^\top x + b_0,
+```
+for any feasible point ``x``. The latter simplifies to ``a_0^\top d < 0``. For
+maximization problems, the inequality is reversed, so that ``a_0^\top d > 0``.
+
+If the solver has found a certificate of dual infeasibility:
+
+ * [`TerminationStatus`](@ref) must be `DUAL_INFEASIBLE`
+ * [`PrimalStatus`](@ref) must be `INFEASIBILITY_CERTIFICATE`
+ * [`VariablePrimal`](@ref) must be the corresponding value of ``d``
```

Comment:
The open question from #433 is what to do with `ConstraintPrimal`. For consistency, it makes the most sense for it to be `A_i d` (i.e., without the constant term `b_i`).

However, I'm not sure if solvers actually support querying this, so it's effort for solver wrappers to implement.

Alternatively, we could just say that querying `ConstraintPrimal` for an infeasibility certificate is undefined behavior.

Reply:
Note for myself tomorrow: it's actually easy to implement if we update the `get_fallback`, so there's no reason not to explicitly define `ConstraintPrimal` here as `A_i d`.

Reply:
Yes, `get_fallback` is the way and the constant should not be there, see also https://github.com/jump-dev/MathOptInterface.jl/issues/1310.

Reply:
Okay done. I still need to add tests for the scalarize and vectorize bridges.